datafusion/sql/src/planner.rs

```diff
@@ -1124,6 +1126,39 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         // process `from` clause
         let plans = self.plan_from_tables(select.from, ctes, outer_query_schema)?;
         let empty_from = matches!(plans.first(), Some(LogicalPlan::EmptyRelation(_)));
+        // build from schema for unqualifier column ambiguous check
+        // we should get only one field for this unqualifier column from schema.
+        let from_schema = {
```

Comment:
Maybe as a follow up PR we could factor this out into a function (as `select_to_plan` is already quite long)

Reply:
> Maybe as a follow up PR we could factor this out into a function (as `select_to_plan` is already quite long)

Thanks! @alamb  I'll propose a follow up PR to handle this.